kernel/interrupts/src/aarch64/table.s

```diff
@@ -0,0 +1,128 @@
+.macro CALL_WITH_CONTEXT handler
+__vector_\handler:
+    // Make room on the stack for the exception context.
+    sub sp,  sp,  #16 * 17
+
+    // Store all general purpose registers on the stack.
+    stp x0,  x1,  [sp, #16 * 0]
+    stp x2,  x3,  [sp, #16 * 1]
+    stp x4,  x5,  [sp, #16 * 2]
+    stp x6,  x7,  [sp, #16 * 3]
+    stp x8,  x9,  [sp, #16 * 4]
+    stp x10, x11, [sp, #16 * 5]
+    stp x12, x13, [sp, #16 * 6]
+    stp x14, x15, [sp, #16 * 7]
+    stp x16, x17, [sp, #16 * 8]
+    stp x18, x19, [sp, #16 * 9]
+    stp x20, x21, [sp, #16 * 10]
+    stp x22, x23, [sp, #16 * 11]
+    stp x24, x25, [sp, #16 * 12]
+    stp x26, x27, [sp, #16 * 13]
+    stp x28, x29, [sp, #16 * 14]
+
+    // Add the exception link register (ELR_EL1), saved program status (SPSR_EL1) and exception
+    // syndrome register (ESR_EL1).
+    mrs x1,  ELR_EL1
+    mrs x2,  SPSR_EL1
+    mrs x3,  ESR_EL1
+
+    stp lr,  x1,  [sp, #16 * 15]
+    stp x2,  x3,  [sp, #16 * 16]
+
+    // x0 is the first argument for the function called through `\handler`.
+    mov x0,  sp
+
+    // Call `\handler`.
+    bl  \handler
+
+    // After returning from exception handling code, replay the saved context and return via
+    // `eret`.
+    b   __exception_restore_context
+
+// .size   __vector_\handler, . - __vector_\handler
+// .type   __vector_\handler, function
+.endm
+
+.macro FIQ_SUSPEND
+1:  wfe
+    b   1b
+.endm
+
+.section .text
+
+// Align by 2^11 bytes, as demanded by ARMv8-A. Same as ALIGN(2048) in an ld script.
+.align 11
```

Comment:
it's not clear to me whether `align` works like this on ARM assembly; i think it matters which assembler we use.

To be more clear, perhaps you should use `.balign 2048`.

Reply:
also it'd be nice to have a link to some docs that explain why such a large alignment is required.

Reply:
`align` is well supported: https://doc.rust-lang.org/nightly/reference/inline-assembly.html

![image](https://user-images.githubusercontent.com/61582713/220607474-38678331-95c0-4cce-b7e3-24af5f6448ab.png)

I'll add an explanation for the alignment requirements

Reply:
Thanks. The extra docs you provide there are great, super clear.

I'd still prefer using ".balign" since 99% of developers reading this in the future will be very puzzled by an `.align 11` statement. Kindly change it to that for the sake of better readability.

Reply:
Ok, done